plugins/orbslam2_plugin/orbslam2_sensors.py

```diff
@@ -0,0 +1,105 @@
+    def __init__(self, rgb_sensor: RGBSensor, depth_sensor: DepthSensor, vocab_file: str, use_slam_viewer: bool, uuid: str = "orbslam", **kwargs: Any):
+        observation_space = self._get_observation_space()
+        super().__init__(**prepare_locals_for_super(locals()))
+        self.rgb_sensor, self.depth_sensor = rgb_sensor, depth_sensor
+        self.vocab_file, self.use_slam_viewer = vocab_file, use_slam_viewer
+        self.slam_system = None
+
+    def _get_observation_space(self):
+        raise NotImplementedError()
+
+    def initialize_slam(self, env):
+        raise NotImplementedError()
+
+    def reset(self, agent_state):
+        if self.slam_system is not None:
+            pose = state_to_pose(agent_state)
+            self.slam_system.reset(pose)
+
+    def stop(self):
+        self.slam_system.shutdown()
+
+
+class ORBSLAMCompassSensorRoboThor(ORBSLAMSensor[RoboThorEnvironment, PointNavTask]):
+
+    def __init__(self, rgb_sensor: RGBSensor, depth_sensor: DepthSensor, vocab_file: str, use_slam_viewer: bool, uuid: str = "target_coordinates_ind", **kwargs: Any):
+        super().__init__(rgb_sensor, depth_sensor, vocab_file, use_slam_viewer, uuid, **kwargs)
+
+    def _get_observation_space(self):
+        return gym.spaces.Box(
+            low=np.finfo(np.float32).min,
+            high=np.finfo(np.float32).max,
+            shape=(2,),
+            dtype=np.float32,
+        )
+
+    def initialize_slam(self, env):
+        settings_file = mkstemp()[1]
+        with open(settings_file, 'w') as file:
+            file.write('%YAML:1.0')
+            slam_settings = init_settings(
+                *[env.last_event.metadata[m] for m in ['screenWidth', 'screenHeight', 'fov']]
+            )
+            file.write(yaml.dump(slam_settings))
+        initial_pose = state_to_pose(env.agent_state())
+        slam_system = orbslam2.SLAM(self.vocab_file, settings_file, 'rgbd', initial_pose, self.use_slam_viewer)
+        os.remove(settings_file)
+        return slam_system
+
+    def get_observation(self, env: RoboThorEnvironment, task: Optional[PointNavTask], *args: Any, **kwargs: Any) -> Any:
+        if self.slam_system is None:
+            self.slam_system = self.initialize_slam(env)
+
+        frame = np.uint8(self.rgb_sensor.get_observation(env, task, *args, **kwargs) * 255)
+        depth = np.float32(self.depth_sensor.get_observation(env, task, *args, **kwargs)[:, :, 0])
+        self.slam_system.track(frame, depth)
+
+        pose = self.slam_system.get_world_pose()
+        agent_position = pose[:3, 3]
+        agent_rotation = np.rad2deg(matrix_to_euler_angles(pose[:3, :3]))
+        rotation_world_agent = GPSCompassSensorRoboThor.quaternion_from_y_angle(agent_rotation[1])
+        goal_position = np.array([task.task_info["target"][k] for k in 'xyz'])
+
+        goal = GPSCompassSensorRoboThor._compute_pointgoal(GPSCompassSensorRoboThor,
+            agent_position, rotation_world_agent, goal_position
+        )
```

Comment:
It seems that `_compute_pointgoal` should be a class or static method? Can you change it so that we don't have to pass `GPSCompassSensorRoboThor` as the first argument? Also, since we're calling `_compute_pointgoal` we probably don't want it to be private, perhaps refactor it to remove the preceding underscore?